crates/legacy/src/builder_state.rs

```diff
@@ -185,6 +185,17 @@ pub struct BuilderState<TYPES: NodeType> {
 
     /// time of next garbage collection for txns
     pub next_txn_garbage_collect_time: Instant,
+
+    /// allow_empty_block_until is a variable that dictates the time until which
+    /// a builder should stop producing empty blocks. This is done specifically
+    /// to allow for faster finalization of previous blocks that have had
+    /// transactions included in them.
+    pub allow_empty_block_until: Option<TYPES::Time>,
+
+    /// allow_empty_block_period is used in conjunction with [TYPES::Time] to
+    /// determine the time period that empty blocks are allowed to be built
+    /// for.
+    pub allow_empty_block_period: u64,
```

Comment:
Nit: Can we make this a `const` rather than a field so that it's always 3 for now? I think this makes the setting clearer and makes it easier to update to 2 later.